tests/50federation/36state.pl

```diff
@@ -532,17 +532,45 @@ sub get_state_ids_from_server {
                Future->done(1);
             }),
          )->then( sub {
+            # the server may send a /state request; be prepared to answer that.
+            # (it may, alternatively, send individual /event requests)
+            my $state_req_fut = $inbound_server->await_request_v1_state(
+               $room_id, $missing_event_y->{event_id},
+            )->then( sub {
```

Comment:
Again, this is just an extension of the existing request for `/state_ids` at Y (which is handled just above), but.

possibly it wants the state just after Y, but there's no way in the protocol to ask for that - instead you have to ask for the state *before* Y, and then add Y yourself, which is what happens here.

We deliberately *avoid* asking for the state before X, because that would mean trusting the remote server to tell us the entire room state. By going round the houses like this, we ensure that we always state-resolve any incoming state (at `Y`) with our own idea of the state (at `B`).  (see also: https://github.com/matrix-org/internal-config/issues/102)

Reply:
Makes sense now! I misread the comment at L350-352 and assumed that the server wouldn't ask for event Y.